pkg/frontend/querymiddleware/codec.go

```diff
@@ -50,6 +52,10 @@ const (
 	totalShardsControlHeader = "Sharding-Control"
 )
 
+func init() {
```

Comment:
I don't think the custom marshalling code you've added belongs to this file. I would move it to `pkg/mimirpb/` package instead.

Reply:
done @ da7291e893b606b8c5ef33fddb3c06a339df702d